src/server/pfs/server/storage_bucket_test.go

```diff
@@ -0,0 +1,125 @@
+	require.NoError(t, os.Setenv("AZURE_STORAGE_KEY", os.Getenv("MICROSOFT_CLIENT_SECRET")))
+	bucketName := os.Getenv("MICROSOFT_CLIENT_CONTAINER")
+	url, err := obj.ParseURL("azblob://" + bucketName)
+	require.NoError(t, err, "should be able to parse url")
+	writeReadDelete(t, url)
+	url, err = obj.ParseURL("wasb://" + bucketName + "@" + os.Getenv("MICROSOFT_CLIENT_ID") + ".blob.core.windows.net")
+	require.NoError(t, err, "should be able to parse url")
+	writeReadDelete(t, url)
+}
+
+func TestGoogle(t *testing.T) {
+	integrationtests.LoadGoogleParameters(t)
+	credFile := path.Join(t.TempDir(), "tmp-google-cred")
+	require.NoError(t, os.WriteFile(credFile, []byte(os.Getenv("GOOGLE_CLIENT_CREDS")), 0666))
+	require.NoError(t, os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", credFile))
+	bucketName := os.Getenv("GOOGLE_CLIENT_BUCKET")
+	url, err := obj.ParseURL("gs://" + bucketName)
+	require.NoError(t, err, "should be able to parse url")
+	writeReadDelete(t, url)
+}
+
+func TestAmazon(t *testing.T) {
+	integrationtests.LoadAmazonParameters(t)
+	require.NoError(t, os.Setenv("AWS_REGION", os.Getenv("AMAZON_CLIENT_REGION")))
+	require.NoError(t, os.Setenv("AWS_SECRET_ACCESS_KEY", os.Getenv("AMAZON_CLIENT_SECRET")))
+	require.NoError(t, os.Setenv("AWS_ACCESS_KEY_ID", os.Getenv("AMAZON_CLIENT_ID")))
+	bucketName := os.Getenv("AMAZON_CLIENT_BUCKET")
+	url, err := obj.ParseURL("s3://" + bucketName)
+	require.NoError(t, err, "should be able to parse url")
+	writeReadDelete(t, url)
+}
+
+func TestGoogleHMAC(t *testing.T) {
+	integrationtests.LoadGoogleHMACParameters(t)
+	require.NoError(t, os.Setenv("AWS_REGION", os.Getenv("GOOGLE_CLIENT_REGION")))
+	require.NoError(t, os.Setenv("AWS_SECRET_ACCESS_KEY", os.Getenv("GOOGLE_CLIENT_HMAC_SECRET")))
+	require.NoError(t, os.Setenv("AWS_ACCESS_KEY_ID", os.Getenv("GOOGLE_CLIENT_HMAC_ID")))
+	require.NoError(t, os.Setenv("CUSTOM_ENDPOINT", "storage.googleapis.com"))
+	bucketName := os.Getenv("GOOGLE_CLIENT_BUCKET")
+	url, err := obj.ParseURL("s3://" + bucketName)
+	require.NoError(t, err, "should be able to parse url")
+	writeReadDelete(t, url)
+}
+
+func TestAmazonECS(t *testing.T) {
+	integrationtests.LoadECSParameters(t)
+	require.NoError(t, os.Setenv("AWS_REGION", "dummy-region"))
+	require.NoError(t, os.Setenv("AWS_SECRET_ACCESS_KEY", os.Getenv("ECS_CLIENT_SECRET")))
+	require.NoError(t, os.Setenv("AWS_ACCESS_KEY_ID", os.Getenv("ECS_CLIENT_ID")))
+	require.NoError(t, os.Setenv("CUSTOM_ENDPOINT", os.Getenv("ECS_CLIENT_CUSTOM_ENDPOINT")))
+	// We may have to check if we have valid ssl for our ECS configuration.
+	require.NoError(t, os.Setenv("DISABLE_SSL", "true"))
+	bucketName := os.Getenv("ECS_CLIENT_BUCKET")
+	url, err := obj.ParseURL("s3://" + bucketName)
+	require.NoError(t, err, "should be able to parse url")
+	writeReadDelete(t, url)
+}
+
+func writeReadDelete(t *testing.T, url *obj.ObjectStoreURL) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*30000)
```

Comment:
That is a long timeout.

Reply:
whoops, I must have left that after introspecting with my debugger